### PR TITLE
Improve the Docker image build configuration

### PR DESCRIPTION
**Pull request by TimMoore**
- Exclude non-implementation projects from publishing.
  This allows `sbt docker:publishLocal` to complete successfully without
  having to list out specific sub-projects.
- Use the `Run` Docker permission strategy, which is more compatible
  with older Docker Engine versions, including the version used by
  OpenShift 3.11.
- Use the AdoptOpenJDK base image.
- Refactor common Docker settings into a method.

### CODE REVIEW

**Comment by TimMoore** [comment]
Marked as WIP because the build is totally broken until #25 is merged. I'll rebase this after it is.

File: build.sbt [hunk 1]
```diff
@@ -41,26 +40,24 @@ lazy val `hello-impl` = (project in file("hello-impl"))
 
 lazy val `hello-proxy-api` = (project in file("hello-proxy-api"))
   .settings(
-    libraryDependencies +=lagomScaladslApi
+    libraryDependencies += lagomScaladslApi,
+    publish in Docker := {}
```
Comment (ignasi35) [inline build.sbt]: I like this explicitly disabling the settings. What do you think about extracting that to:

```scala
def disableDockerPublish(): Seq[Setting[_]] = 
  Seq(
    publish in Docker := {}
  )
}
```

**Review by ignasi35 — COMMENTED** [review]
Nice improvements!

I added a small comment with an idea, let me know what you think.

File: build.sbt (hunk 1 above)
Comment (TimMoore) [reply build.sbt]: Yes, I like it. Makes it much more obvious what that cryptic bit of configuration is accomplishing.

**Review by ignasi35 — APPROVED** [review]

**Comment by ignasi35** [comment]
Y U no mergy @mergify ?

**Comment by TimMoore** [comment]
It didn't merge because it's looking for "Travis CI - Pull Request" instead of "Travis CI - Branch". I switched them yesterday.

**Pull request merged**